packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -275,200 +208,114 @@ export const getOrderFilledEvents = async (
   const currentBlockNumber = await params.provider.getBlockNumber();
   const maxBlockLookBack = params.maxBlockLookBack;
 
-  return Promise.all(
-    markets.map(async (market) => {
-      const searchConfig = {
-        fromBlock: market.eventBlockNumber,
-        toBlock: currentBlockNumber,
-        maxBlockLookBack,
-      };
+  const searchConfig = {
+    fromBlock: startBlockNumber,
+    toBlock: currentBlockNumber,
+    maxBlockLookBack,
+  };
 
-      const events: Event[] = await paginatedEventQuery(
-        ctfExchange,
-        ctfExchange.filters.OrderFilled(null, null, null, null, null, null, null, null),
-        searchConfig
-      );
-
-      const outcomeTokenOne = await Promise.all(
-        events
-          .filter((event) => {
-            return [event?.args?.takerAssetId.toString(), event?.args?.makerAssetId.toString()].includes(
-              market.clobTokenIds[0]
-            );
-          })
-          .map((event) => getTradeInfoFromOrderFilledEvent(params.provider, event))
-      );
-
-      const outcomeTokenTwo = await Promise.all(
-        events
-          .filter((event) => {
-            return [event?.args?.takerAssetId.toString(), event?.args?.makerAssetId.toString()].includes(
-              market.clobTokenIds[1]
-            );
-          })
-          .map((event) => getTradeInfoFromOrderFilledEvent(params.provider, event))
-      );
+  const events: Event[] = await paginatedEventQuery(
+    ctfExchange,
+    ctfExchange.filters.OrderFilled(null, null, null, null, null, null, null, null),
+    searchConfig
+  );
 
-      return {
-        ...market,
-        orderFilledEvents: [outcomeTokenOne, outcomeTokenTwo],
-      };
-    })
+  const outcomeTokenOne = await Promise.all(
+    events
+      .filter((event) => {
+        return [event?.args?.takerAssetId.toString(), event?.args?.makerAssetId.toString()].includes(clobTokenIds[0]);
+      })
+      .map((event) => getTradeInfoFromOrderFilledEvent(params.provider, event))
+  );
+
+  const outcomeTokenTwo = await Promise.all(
+    events
+      .filter((event) => {
+        return [event?.args?.takerAssetId.toString(), event?.args?.makerAssetId.toString()].includes(clobTokenIds[1]);
+      })
+      .map((event) => getTradeInfoFromOrderFilledEvent(params.provider, event))
   );
+
+  return [outcomeTokenOne, outcomeTokenTwo];
 };
 
-const loadAbi = (filename: string): any => {
-  return require(`./abi/${filename}.json`);
+export const calculatePolymarketQuestionID = (ancillaryData: string): string => {
+  return ethers.utils.keccak256(ancillaryData);
```

Comment:
This is really how the compute question id? I had no idea it was this simple.

Great find!